src/DN2.jl

```diff
@@ -2,106 +2,211 @@ module DN2
+        ts, pts = bezier_polyline(ctrls, samples)
+        for i in 1:samples-2
+            p1, p2 = pts[i], pts[i+1]
+            for j in i+2:samples-1  # Preskoči sosednje ali prekrivajoče
+                q1, q2 = pts[j], pts[j+1]
+                result = segment_intersection(p1, p2, q1, q2)
+                if result !== nothing
+                    (u1, u2) = result
+                    t1 = (1 - u1)*ts[i] + u1*ts[i+1]
+                    t2 = (1 - u2)*ts[j] + u2*ts[j+1]
+                    return t1, t2
+                end
+            end
         end
-
-        return result
+        return nothing
     end
 
-    """Izvede polinomsko seštevanje polinomov p in q."""
-    function poly_add(p, q)
-        len = max(length(p), length(q))
-        r = zeros(Float64, len)
-        for i in eachindex(p)
-            r[i] += p[i]
-        end
-        for i in eachindex(q)
-            r[i] += q[i]
+    """
+    Iz začetnih t1 in t2 z bisekcijo izboljša oceno presečišča.
+    """
+    function refine_intersection(ctrls::Vector{Vector{Float64}}, t1::Float64, t2::Float64;
+                                tol::Float64 = 1e-10, max_iter::Int = 100)
+
+        function refine_single(fixed_t, lo, hi, fixed_is_t1)
+            for _ in 1:max_iter
+                mid = (lo + hi) / 2
+                p_fixed = evaluate_bezier(ctrls, fixed_t)
+                p_mid   = evaluate_bezier(ctrls, mid)
+                dist_lo = norm(p_fixed .- evaluate_bezier(ctrls, lo))
+                dist_hi = norm(p_fixed .- evaluate_bezier(ctrls, hi))
+                dist_mid = norm(p_fixed .- p_mid)
+
+                if dist_lo < dist_hi
+                    hi = mid
+                else
+                    lo = mid
+                end
+
+                if abs(hi - lo) < tol
+                    return (lo + hi) / 2
+                end
+            end
+            return (lo + hi) / 2
         end
-        return r
-    end
 
-    """Potencira polinom p na stopnjo n."""
-    function poly_pow(p, n)
-        result = [1.0]
-        for _ in 1:n
-            result = poly_mul(result, p)
-        end
-        return result
-    end
+        # Izboljšaj t1 pri fiksnem t2
+        t1 = refine_single(t2, t1 - 0.01, t1 + 0.01, true)
+        # Izboljšaj t2 pri fiksnem t1
+        t2 = refine_single(t1, t2 - 0.01, t2 + 0.01, false)
 
-    """Izračuna odvod polinoma p."""
-    function poly_deriv(p)
-        n = length(p)
-        return [p[i] * (i - 1) for i in 2:n]
+        return t1, t2
     end
 
-    """Zmnnoži Bezierjeva polinoma za x in y, in nato odšteje enega od drugega."""
-    function cross_term(xpoly, ypoly)
-        dx = poly_deriv(xpoly)
-        dy = poly_deriv(ypoly)
-        term1 = poly_mul(xpoly, dy)
-        term2 = poly_mul(ypoly, dx)
-        return poly_add(term1, -1 .* term2)
-    end
+    """
+    Izračuna predznačeno ploščino po Greeovem teoremu.
+    """
+    function signed_area(ctrls::Vector{Vector{Float64}}; a=0.0, b=1.0)
```

Comment:
Sam bi to funkcijo razbil na dva dela. Splošno funkcijo, ki računa integral z GL kvadraturo na danem intervalu in funkcijo, ki izračuna integrand za dano Bézierovo krivuljo. Tako bi lahko posebej testirali posebej implementacijo GL in integranda.